src/assignments/tech-jobs-oo.rst

```diff
@@ -1,6 +1,586 @@
+- Full Stack”. Each job that wants to use that position type holds onto a
+reference to the given object. Similarly, we can have one ``Employer`` object
+for each employer.
+
+Aside from reducing the amount of raw data / memory that the application uses,
+this will allow data to be updated more easily and properly. If we need to
+change the name of an employer (e.g. due to a typo or a name change at the
+company), we can change it in one place---the single ``Employer`` object that
+represents that company.
+
+Enable Extension
+~~~~~~~~~~~~~~~~~
+
+While the four ``Job`` properties represented by objects will primarily be used
+for their string values, it’s easy to imagine adding new properties to address
+future needs.
+
+For example, it would be useful for an ``Employer`` object to have an address,
+a primary contact, and a list of jobs available at that employer.
+
+For a ``Location`` object, useful information includes a list of zip codes
+associated with that location, in order to determine the city and state for an
+employer or job.
+
+If we were to store these properties as strings, extending and modifying this
+behavior would be much more complicated and difficult moving forward.
+
+Your Assignment
+---------------
+
+You’ve been assigned the following tasks:
+
+#. Review Sally's code in the ``Employer`` class to learn how to assign a
+   unique ID.
+#. Add getters, setters, and custom methods as needed to the ``Location``,
+   ``CoreCompetency``, and ``PositionType`` classes.
+#. Complete the ``Job`` class using what you learned in steps 1 and 2.
+#. Use unit testing to verify the constructors and ``equals`` methods for the
+   ``Job`` class.
+#. Use TDD to design and code a custom ``toString`` method.
+#. Use inheritance to DRY the code within ``Employer``, ``Location``,
+   ``CoreCompetency``, and ``PositionType``.
+
+1) Explore the ``Employer`` Class
+----------------------------------
+
+Open the ``Employer`` file in IntelliJ and examine the code. In addition to the
+two fields---``id`` and ``value``---the class includes the standard getters and
+setters as well as some custom methods like ``toString`` and ``equals``.
+
+You can refer to these examples as you fill in the missing pieces in the other
+classes, but for now let's take a closer look at the constructors.
+
+Assign a Unique ID
+^^^^^^^^^^^^^^^^^^^
+
+One neat trick we can use is to automatically assign each new object a unique
+ID number.
+
+.. admonition:: Example
+
+   Examine the two constructors in ``Employer.java``:
+
+   .. sourcecode:: java
+      :linenos:
+
+      public class Employer {
+         private int id;
+         private static int nextId = 1;
+         private String value;
+
+         public Employer() {
+            id = nextId;
+            nextId++;
+         }
+
+         public Employer(String aValue) {
+            this();
+            this.value = aValue;
+         }
+
+         // Getters and setters omitted from this view.
```

Comment:
In this class, as well as all others using an `id` field, `setId` should be `private`.

Reply:
I agree, but this complicates one of the tests.  Let's chat about this one when we can both look at the completed code.

Reply:
I revised the Test ``equals`` section.  Let me know what you think about the changes.